### PR TITLE
Add guard.proto

### PR DESCRIPTION
**Pull request by JustinGaoF**

### CODE REVIEW

File: protos/core/Btfs.proto [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+	int64 withhold_amount = 8;
+	int64 paid_amount = 9;
+	int64 next_payment_amount = 10;
+	Timestamp next_payment_time = 11;
+	Timestamp last_payment_time = 12;
+	Timestamp sign_time = 13;
+	bytes signature = 14;
+}
+
+message UpdatePaymentRequest{
```
Comment (jin-tron) [inline protos/core/Btfs.proto]: By create a new escrow, escrow don't need this UpdatePaymentRequest

File: protos/core/Btfs.proto [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+	status status = 4;
+	bytes payment_id = 5; //escrow assigned id for the payment
+	int64 total_amount = 6; //whole amount of the subcontract , equal rent amount + collateral amount
+	int64 collateral_amount = 7;
+	int64 withhold_amount = 8;
+	int64 paid_amount = 9;
+	int64 next_payment_amount = 10;
+	Timestamp next_payment_time = 11;
+	Timestamp last_payment_time = 12;
+	Timestamp sign_time = 13;
```
Comment (jin-tron) [inline protos/core/Btfs.proto]: These info btfs guard already have, better don't need escrow to return when create escrow.

File: protos/core/Btfs.proto [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+	bytes signature =4; //requester signed 
+}
+
+message StartEscrowRequest{
+	ProofOfFund escrow_amount = 1; // the proof of renter had reserve money in the escrow for such chunk 
+	ProofOfFund escrow_collateral_amount = 2; //the proof of host for collateral
+	int pay_frequency = 3; 
+	Timestamp start_time = 4;
+	Timestamp end_time = 5;
+	bytes peerId = 6; //guard's instance's peerId
```
Comment (jin-tron) [inline protos/core/Btfs.proto]: Escrow don't need the peerId

File: protos/core/Btfs.proto [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+	bytes file_hash = 1;
+	bytes renter_address = 2;
+	bytes requester_address = 3;
+	bytes signature =4; //requester signed 
+}
+
+message StartEscrowRequest{
+	ProofOfFund escrow_amount = 1; // the proof of renter had reserve money in the escrow for such chunk 
+	ProofOfFund escrow_collateral_amount = 2; //the proof of host for collateral
+	int pay_frequency = 3;
```
Comment (jin-tron) [inline protos/core/Btfs.proto]: Need review escrow payment  proto, by now escrow support "schedule" and "payout_times", don't count payout times from the period. That means need caller to give escrow how many times.

File: protos/core/Btfs.proto [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+	ProofOfFund escrow_amount = 1; // the proof of renter had reserve money in the escrow for such chunk 
+	ProofOfFund escrow_collateral_amount = 2; //the proof of host for collateral
+	int pay_frequency = 3; 
+	Timestamp start_time = 4;
+	Timestamp end_time = 5;
+	bytes peerId = 6; //guard's instance's peerId
+	bytes signature = 7; 
+}
+
+message PaymentStatus{
```
Comment (jin-tron) [inline protos/core/Btfs.proto]: Does guard need to know the current payment status of an escrow? If yes, then the details are useful, otherwise for create escrow return value, we don't need many details as below comment.
If guard need query escrow payment status, we need add a interface in escrow for this query.

File: protos/core/Btfs.proto [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+//host provided service
+service host{
+	rpc challenge(ProofOfReplicateChallenge) returns (ProofOfReplicateChallenge){}
+}
+
+
+
+
+message ProofOfFund{
+	bytes id = 1;// the proof id provide by escrow
```
Comment (jin-tron) [inline protos/core/Btfs.proto]: this is int64

File: protos/core/Btfs.proto [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+	bytes reference_id = 2; //the reference id to present purpose of such proof, in the BTFS, it is file_hash or chunk_hash
+	bytes escrow_address = 3; //the escrow service’s public key 
+	bytes owner_address = 4; //owner’s public key
+	bytes target_address = 5; //payment target’s pub key, in BTFS, it is host pub key for renter prepare such POF, and is renter pub key for host to prepare such POF
```
Comment (jin-tron) [inline protos/core/Btfs.proto]: This will not filled by escrow

File: protos/core/Btfs.proto (hunk 4 above)
Comment (JustinGaoF) [reply protos/core/Btfs.proto]: Will change to escrow defined payout schedule format.

File: protos/core/Btfs.proto (hunk 7 above)
Comment (JustinGaoF) [reply protos/core/Btfs.proto]: This field will be filled by the renter, and from my understanding, only id, signed_time, escrow_signature were filled by escrow service, not sure we can meet understanding.

File: protos/core/Btfs.proto [hunk 8]
```diff
@@ -0,0 +1,256 @@
+syntax = "proto3";
+
+option java_multiple_files = true;
+option java_package = "org.tron.protos"; //Specify the name of the package that generated the Java file
+option java_outer_classname = "Protocol"; //Specify the class name of the generated Java file
```
Comment (taiyangc) [inline protos/core/Btfs.proto]: not sure we really need these lines.

File: protos/core/Btfs.proto [hunk 9]
```diff
@@ -0,0 +1,256 @@
...
+
+
+package btfs;
+
+
+service guard {
```
Comment (taiyangc) [inline protos/core/Btfs.proto]: Before we finalize, please make sure the naming follows standard protobuf conventions: https://developers.google.com/protocol-buffers/docs/style

Comment (taiyangc) [reply protos/core/Btfs.proto]: I think there are some tools online to help with this

File: protos/core/Btfs.proto [hunk 10]
```diff
@@ -0,0 +1,230 @@
+syntax = "proto3";
+
+
+
+package btfs;
```
Comment (taiyangc) [inline protos/core/Btfs.proto]: Per discussion with @wxue and @jin-tron , we will put all .proto files under `protos/` directly. This file should be renamed guard.proto

The package name should be shared as well.

File: protos/core/Btfs.proto [hunk 11]
```diff
@@ -0,0 +1,230 @@
...
+
+// service EscrowService{. //FYI from Cong's review, this part will be removed eventually
+// 	submitContracts()
+// 	fullPay()
+// 	checkContract()
+// 	cancelContract()
+// }
+
+service GuardService{
+	rpc submitFileStoreMeta(FileStoreMeta) returns (Result){}
```
Comment (taiyangc) [inline protos/core/Btfs.proto]: what is the `{}` for

File: protos/core/Btfs.proto [hunk 12]
```diff
@@ -0,0 +1,230 @@
...
+	rpc health() returns(HealthStatus){} //used for other nodes check guard health , TBD input argument
+  	rpc sendQuestions(Questions) returns(Result){} //used for renter send the questions to guard
+  	rpc checkFileStoreMeta(CheckFileStoreMetaRequest) returns (FileStoreMeta){} //reserved for other nodes to check contract status
+}
+
+service HostService{
```
Comment (taiyangc) [inline protos/core/Btfs.proto]: Remove. This will be on libp2p and thus will be an http api.

File: protos/core/Btfs.proto [hunk 13]
```diff
@@ -0,0 +1,230 @@
...
+	rpc challenge(ProofOfReplicateChallenge) returns (ProofOfReplicateChallenge){} //used for receieve challenge question and return the answer, TBD seems can be removed as the BTFS has embed function
```
Comment (taiyangc) [inline protos/core/Btfs.proto]: You can keep `ProofOfReplicateChallenge` message, but this function will not be used.

File: protos/core/Btfs.proto [hunk 14]
```diff
@@ -0,0 +1,230 @@
...
+	rpc cancelContract(Contract) returns (Result){}
```
Comment (taiyangc) [inline protos/core/Btfs.proto]: Won't be supported by MVP.

File: protos/core/Btfs.proto [hunk 15]
```diff
@@ -0,0 +1,230 @@
...
+	bytes host_address = 3; //host’s wallet address
+	bytes shard_hash = 4; //shard’s hash
+	int shard_index = 5; //shard’s sequence of the original file
+	int64 shard_file_size = 6; //shard’s file size
+	bytes file_hash = 7; //original file's hash, reserved point back to FileStoreMeta
+	Timestamp rent_start = 7; //rent start time
+	Timestamp rent_end = 8 ; //rent end time
+	bytes guard_address = 9; //renter proposed guard's address
+	bytes escrow_address = 10; //renter proposed escrow's address
+	bytes renter_propose_signature = 11; //renter signature for proposed information, signed for above information
```
Comment (taiyangc) [inline protos/core/Btfs.proto]: mention sha2-256 for hashing

File: protos/core/Btfs.proto [hunk 16]
```diff
@@ -0,0 +1,230 @@
...
+	int64 agreed_price = 12; //the price which agreed by host and rental, on the first phase, it equal to the proposed_price
+	int64 agreed_amount = 13; //the amount which agreed by host and rental, on the first phase, it should be equal file_size*duration*agreed_price
```
Comment (taiyangc) [inline protos/core/Btfs.proto]: It will always be `file_size*duration*agreed_price`, so is this really necessary?

File: protos/core/Btfs.proto [hunk 17]
```diff
@@ -0,0 +1,230 @@
...
+	int64 agreed_collaeteral_amount = 14; 
+	enum Schedule{
```
Comment (taiyangc) [inline protos/core/Btfs.proto]: define enum outside message.

File: protos/core/Btfs.proto [hunk 18]
```diff
@@ -0,0 +1,230 @@
...
+    	QUARTERLY = 1;
+    	ANNUALLY = 2;
+	}
+    Schedule payout_schedule = 15; // schedule for payouts
+   	int32 num_payouts = 16; // numbers of payouts
+	Timestamp escrow_signed_time = 17; //contract signed time
+	bytes renter_signature = 18; //renter’s signature for the sub contract
+	bytes host_signature = 19; //host’s signature from the sub contract
+	bytes escrow_signature = 20； //escrow's signature 
+	enum State{
```
Comment (taiyangc) [inline protos/core/Btfs.proto]: define enum outside message.

File: protos/core/Btfs.proto [hunk 19]
```diff
@@ -0,0 +1,230 @@
...
+	int64 agreed_price = 12; //the price which agreed by host and rental, on the first phase, it equal to the proposed_price
+	int64 agreed_amount = 13; //the amount which agreed by host and rental, on the first phase, it should be equal file_size*duration*agreed_price
+	int64 agreed_collaeteral_amount = 14; 
+	enum Schedule{		
+    	MONTHLY = 0;
+    	QUARTERLY = 1;
+    	ANNUALLY = 2;
+	}
+    Schedule payout_schedule = 15; // schedule for payouts
+   	int32 num_payouts = 16; // numbers of payouts
```
Comment (taiyangc) [inline protos/core/Btfs.proto]: This is pre-calculated? `ceil(duration/schedule)`?

File: protos/core/Btfs.proto [hunk 20]
```diff
@@ -0,0 +1,230 @@
...
+	Timestamp escrow_signed_time = 17; //contract signed time
+	bytes renter_signature = 18; //renter’s signature for the sub contract
```
Comment (taiyangc) [inline protos/core/Btfs.proto]: What's difference between `renter_propose_signature` and `renter_signature`?

File: protos/core/Btfs.proto [hunk 21]
```diff
@@ -0,0 +1,230 @@
...
+	bytes escrow_signature = 20； //escrow's signature 
+	enum State{
+		DRAFT = 0; //the contract was draft, but the escrow service did not received the money
+		SIGNED = 1;  // the escrow service received the money from renter, the contract was start.
+		UPLOADED = 2; //the first challenge was done, which means the shard was in the host, all pending payouts can be issued now
+		LOST = 3; // the challenge failed, the following payout should be hold or even stopped
+		CANCELED = 4; //if the shard was rebuild after LOST, or after some days of NOQUESTION, it go to CANCELED status
+		CLOSED = 5; // the all work in such contract was done
+	}
+	State state = 21;
```
Comment (taiyangc) [inline protos/core/Btfs.proto]: I prefer moving `state` a bit on top of the definition because it determines which fields may be valid during which state. Currently the only downside of this design is that this is not very clear - let's discuss

File: protos/core/Btfs.proto [hunk 22]
```diff
@@ -0,0 +1,230 @@
...
+	bytes guard_signature = 24;
+}
+
+
+message FileStoreMeta{
+	bytes renter_address = 1; // buyer wallet address
+	bytes file_hash = 2; // file’s hash
+	int64 file_size = 3; // file’s size
+	Timestamp rent_start = 4; //rent start time
+	Timestamp rent_end = 5; //rent end time
```
Comment (taiyangc) [inline protos/core/Btfs.proto]: Seems some shared fields here. We should define all shared fields in a separate `message` and have both `FileStoreMeta` and individual `Contract` use the same type for finer control.

File: protos/core/Btfs.proto [hunk 23]
```diff
@@ -0,0 +1,230 @@
...
+	int check_frequence = 6; // shard check frequence in a year,  1 means check once per year
```
Comment (taiyangc) [inline protos/core/Btfs.proto]: I prefer defining "DAILY, WEEKLY, MONTHLY" etc.

File: protos/core/Btfs.proto [hunk 24]
```diff
@@ -0,0 +1,230 @@
...
+	int recover_threshold = 11; //reserved field, indicate the threshold for starting to recover
+	repeated Contract Contracts = 12;
+	bytes escrow_address = 13;
+	bytes guard_address = 14;
+	bytes renter_signature = 15; //renter's signature for above information
+	Timestamp guard_receive_time = 16; //guard received time, which used to calculate the timeout for cancel
+	bytes change_log = 17; //TBD one field to store the contract change log, need to discuss whether need the field
+	Timestamp last_modify_time = 18;
+	bytes guard_signature = 19;
+	//TBD need to add one more field for status?
```
Comment (taiyangc) [inline protos/core/Btfs.proto]: Yes, even if it's reserved field for now.

File: protos/core/Btfs.proto [hunk 25]
```diff
@@ -0,0 +1,230 @@
...
+	Timestamp last_modify_time = 18;
+	bytes guard_signature = 19;
+	//TBD need to add one more field for status? 
+}
+
+message ChanllengeQuestion{
+	bytes shard_hash = 1 //shard hash 
+	bytes host_address = 2 //host's address
+	int chunk_index = 3 //the chunk's index for calculation
+	int random_nounce = 4 //the challenge questions's random nounce
```
Comment (taiyangc) [inline protos/core/Btfs.proto]: -> nonce

File: protos/core/Btfs.proto [hunk 26]
```diff
@@ -0,0 +1,230 @@
...
+
+message ChanllengeQuestion{
+	bytes shard_hash = 1 //shard hash 
+	bytes host_address = 2 //host's address
+	int chunk_index = 3 //the chunk's index for calculation
+	int random_nounce = 4 //the challenge questions's random nounce 
+	bytes expect_answer = 5 //the challenge expected answer
+}
+
+message ProofOfReplicateChallenge{
```
Comment (taiyangc) [inline protos/core/Btfs.proto]: -> ProofOfReplication

File: protos/core/Btfs.proto [hunk 27]
```diff
@@ -0,0 +1,230 @@
...
+	ChanllengeQuestion question = 1; //of course expect_answer should be clear before send to the host`
```
Comment (taiyangc) [inline protos/core/Btfs.proto]: Make `expect_answer` an optional if it needs to be cleared

File: protos/core/Btfs.proto [hunk 28]
```diff
@@ -0,0 +1,230 @@
...
+	Timestamp challenge_time = 2 //challenge time from guard
+	bytes guard_address = 3 //guar's address
+	bytes guard_signature = 4 //guard signed for the challenge
+	bytes host_answer = 5 //challenge answer
+	Timestamp host_sign_time = 6 //response time from host
```
Comment (taiyangc) [inline protos/core/Btfs.proto]: In general, response should be a different struct, with the above included.

File: protos/core/Btfs.proto [hunk 29]
```diff
@@ -0,0 +1,230 @@
...
+
+
+message ChallengeQuestions{
+	bytes file_hash = 1; //file's hash
+	bytes shard_hash = 2;
+	bytes preparer_address = 3 ; //challenge questions preparer's address, it is renter address at phase 1
+	int request_count = 4;
+	repeated ChanllengeQuestion = 5;
+	Timestamp prepare_time = 6;
+	bytes renter_signature = 7;
```
Comment (taiyangc) [inline protos/core/Btfs.proto]: -> `preparer_signature`

File: protos/core/Btfs.proto [hunk 30]
```diff
@@ -0,0 +1,230 @@
...
+
+message CheckFileStoreMetaRequest{
+	bytes file_hash = 1;
+	bytes renter_address = 2;
+	bytes requester_address = 3;
+	bytes signature =4; //requester signed 
+}
+
+message Result {
+    enum response_code {
```
Comment (taiyangc) [inline protos/core/Btfs.proto]: make enum public

File: protos/core/Btfs.proto [hunk 31]
```diff
@@ -0,0 +1,230 @@
...
+	bytes guard_signature = 4 //guard signed for the challenge
+	bytes host_answer = 5 //challenge answer
+	Timestamp host_sign_time = 6 //response time from host
+	bytes host_signature = 7
+}
+
+
+message ChallengeQuestions{
+	bytes file_hash = 1; //file's hash
+	bytes shard_hash = 2;
```
Comment (taiyangc) [inline protos/core/Btfs.proto]: challenge questions per file per shard? so you need to store 30 challengequestions per *FileStoreMeta*? Isn't it easier to include in the struct directly?

File: protos/core/Btfs.proto [hunk 32]
```diff
@@ -0,0 +1,230 @@
...
+      SUCCESS = 0;
+      SIGNATURE_FORMAT_ERROR = 1;
+      COMPUTE_ADDRESS_ERROR = 2;
+      OTHER_ERROR = 20;
+    }
+    response_code code = 1;
+    string message = 2;
+}
+
+message HealthStatus{ //TBD if we have another defined object for it, use that object
```
Comment (taiyangc) [inline protos/core/Btfs.proto]: @wxue @jin-tron @JustinGaoF this needs to be in a shared proto shared by all services.

File: protos/core/Btfs.proto [hunk 33]
```diff
@@ -0,0 +1,230 @@
...
+	bytes peerID =1 //guard’s peer ID
```
Comment (taiyangc) [inline protos/core/Btfs.proto]: peer_id

File: protos/core/Btfs.proto [hunk 34]
```diff
@@ -0,0 +1,230 @@
...
+	bytes guard_address = 2 //guard’s pub key/address
+	enum status{
+		WRONG = 0;
+		RUNNING = 1;
+		BOOTSTRAP = 2; //reserve 
+		PARTIAL_STOP = 3; //reserve field, some function cannot work
```
Comment (taiyangc) [inline protos/core/Btfs.proto]: @wxue @jin-tron @JustinGaoF please sync on all possible status codes, make them generic if possible

Comment (taiyangc) [reply protos/core/Btfs.proto]: If you need specific status, you can always have an `extra_info` field for api custom data

File: protos/core/Guard.proto [hunk 35]
```diff
@@ -0,0 +1,230 @@
+syntax = "proto3";
+
+
+
+package btfs;
```
Comment (wxue) [inline protos/core/Guard.proto]: I think should standardize it by adding
```option go_package = "xxxxpb";```

File: protos/core/Btfs.proto (hunk 34 above)
Comment (wxue) [reply protos/core/Btfs.proto]: sounds good, we probably need to take the health message outside of this as a more shared proto

**Comment by wxue** [comment]
overall speaking, I'd recommend using a protobuf linter

File: protos/core/Guard.proto [hunk 36]
```diff
@@ -0,0 +1,230 @@
...
+	bytes guard_address = 14;
+	bytes renter_signature = 15; //renter's signature for above information
+	Timestamp guard_receive_time = 16; //guard received time, which used to calculate the timeout for cancel
+	bytes change_log = 17; //TBD one field to store the contract change log, need to discuss whether need the field
+	Timestamp last_modify_time = 18;
+	bytes guard_signature = 19;
+	//TBD need to add one more field for status? 
+}
+
+message ChanllengeQuestion{
```
Comment (taiyangc) [inline protos/core/Guard.proto]: challengequestion

File: protos/core/Btfs.proto (hunk 22 above)
Comment (taiyangc) [reply protos/core/Btfs.proto]: optional

**Comment by taiyangc** [comment]
@wxue please share linters with others if you find a good one

File: protos/escrow.proto [hunk 37]
```diff
@@ -0,0 +1,99 @@
+syntax = "proto3";
```
Comment (taiyangc) [inline protos/escrow.proto]: Can you remove this commit from junlin's PR thx

Comment (jin-tron) [reply protos/escrow.proto]: removed

File: protos/core/Guard.proto [hunk 38]
```diff
@@ -0,0 +1,178 @@
+syntax = "proto3";
+
+
+
+package btfs;
+
+
+
+service GuardService{
+	rpc submitFileStoreMeta(FileStoreStatus) returns (Result){}
```
Comment (taiyangc) [inline protos/core/Guard.proto]: Upper case `SubmitFileStoreMeta`

File: protos/core/Guard.proto [hunk 39]
```diff
@@ -0,0 +1,178 @@
...
+      SUCCESS = 0;
+      SIGNATURE_FORMAT_ERROR = 1;
+      COMPUTE_ADDRESS_ERROR = 2;
+      OTHER_ERROR = 20;
+    }
+    response_code code = 1;
+    string message = 2;
+}
+
+message HealthStatus{
```
Comment (taiyangc) [inline protos/core/Guard.proto]: Move to different file, please talk with weiyu and jin

File: protos/core/Guard.proto (hunk 36 above)
Comment (taiyangc) [reply protos/core/Guard.proto]: Still a typo here.

File: protos/core/Guard.proto [hunk 40]
```diff
@@ -0,0 +1,178 @@
...
+	bytes renter_signature = 3; //renter’s signature for the sub contract
+	bytes host_signature = 4; //host’s signature from the sub contract
+	Timestamp escrow_signed_time = 5; //contract signed time
+	bytes escrow_signature = 6； //escrow's signature 
+	repeated log change_log = 7; //TBD one field to store the contract change log, need to discuss whether need the field
+	Timestamp last_modify_time = 8;
+	bytes guard_signature = 9;
+}
+
+message log{
```
Comment (taiyangc) [inline protos/core/Guard.proto]: Upper case `Log`

File: protos/core/Guard.proto [hunk 41]
```diff
@@ -0,0 +1,178 @@
...
+	Timestamp changeTime =1;
```
Comment (taiyangc) [inline protos/core/Guard.proto]: change_time

File: protos/core/Btfs.proto (hunk 8 above)
Comment (JustinGaoF) [reply protos/core/Btfs.proto]: removed as need not.

File: protos/core/Btfs.proto (hunk 9 above)
Comment (JustinGaoF) [reply protos/core/Btfs.proto]: done the change for the format using lint provide by weiyu

File: protos/core/Btfs.proto (hunk 10 above)
Comment (JustinGaoF) [reply protos/core/Btfs.proto]: Done the change

File: protos/guard/Guard.proto [hunk 42]
```diff
@@ -0,0 +1,192 @@
...
+
+
+service GuardService{
+	rpc SubmitFileStoreMeta(FileStoreStatus) returns (Result){}
+	//rpc health() returns(HealthStatus){} //used for other nodes check guard health , TBD input argument discuss with Weiyu & Jin
+  	rpc SendQuestions(ChallengeQuestions) returns(Result){} //used for renter send the questions to guard
+  	rpc CheckFileStoreMeta(CheckFileStoreMetaRequest) returns (FileStoreStatus){} //reserved for other nodes to check contract status
+}
+
+message Timestamp {
```
Comment (taiyangc) [inline protos/guard/Guard.proto]: Escrow PR was using google grpc's timestamp? Could you sync to be the same?

File: protos/guard/Guard.proto [hunk 43]
```diff
@@ -0,0 +1,192 @@
...
+	}
+	MetaState state = 2;
+	repeated Contract contracts = 3;
+	bytes renter_signature = 4; //renter's signature for FileStoreMeta and all contractMeta
+	Timestamp guard_receive_time = 5; //guard received time, which used to calculate the timeout for cancel
+	repeated Log change_log = 6;
+	bytes guard_signature = 7;
+}	
+
+message ChanllengeQuestion{
```
Comment (taiyangc) [inline protos/guard/Guard.proto]: This typo still needs correction

File: protos/guard/Guard.proto [hunk 44]
```diff
@@ -0,0 +1,192 @@
+syntax = "proto3";
```
Comment (taiyangc) [inline protos/guard/Guard.proto]: Filename should be lowercase `guard.proto` ... I thought linter should do that. Weiyu is taking a look at linter again.

File: protos/core/Btfs.proto (hunk 12 above)
Comment (JustinGaoF) [reply protos/core/Btfs.proto]: done the removal

File: protos/core/Btfs.proto (hunk 13 above)
Comment (JustinGaoF) [reply protos/core/Btfs.proto]: As we discussed, we got agreement on it.

File: protos/core/Btfs.proto (hunk 14 above)
Comment (JustinGaoF) [reply protos/core/Btfs.proto]: Done the removal

File: protos/core/Btfs.proto (hunk 31 above)
Comment (JustinGaoF) [reply protos/core/Btfs.proto]: Add new message FileChallengeQuestions to group all shard questions.

File: protos/core/Btfs.proto (hunk 33 above)
Comment (JustinGaoF) [reply protos/core/Btfs.proto]: Done the change

File: protos/core/Guard.proto (hunk 36 above)
Comment (JustinGaoF) [reply protos/core/Guard.proto]: solved, thanks for finding.

File: protos/core/Guard.proto (hunk 40 above)
Comment (JustinGaoF) [reply protos/core/Guard.proto]: Solved

File: protos/core/Guard.proto (hunk 38 above)
Comment (JustinGaoF) [reply protos/core/Guard.proto]: Solved.

File: protos/guard/Guard.proto (hunk 44 above)
Comment (JustinGaoF) [reply protos/guard/Guard.proto]: Solved.

File: protos/core/Btfs.proto (hunk 15 above)
Comment (JustinGaoF) [reply protos/core/Btfs.proto]: the field was removed at current process

File: protos/guard/Guard.proto (hunk 42 above)
Comment (JustinGaoF) [reply protos/guard/Guard.proto]: Done the change. Actually my timestamp was copied from google too, it is good for can just use reference.

File: protos/guard/Guard.proto [hunk 45]
```diff
@@ -0,0 +1,192 @@
...
+    	MONTHLY = 0;
+    	QUARTERLY = 1;
+    	ANNUALLY = 2;
+	}
+    Schedule payout_schedule = 15; // schedule for payouts
+   	int32 num_payouts = 16; // numbers of payouts
+}
+
+message Contract{ //the contract executor record, changed frequently after proposal
+	Contract contract = 1;
```
Comment (jin-tron) [inline protos/guard/Guard.proto]: should be ContractMeta as I guess

File: protos/guard/Guard.proto [hunk 46]
```diff
@@ -0,0 +1,192 @@
...
+		UPLOADED = 2; //the first challenge was done, which means the shard was in the host, all pending payouts can be issued now
+		LOST = 3; // the challenge failed, the following payout should be hold or even stopped
+		CANCELED = 4; //if the shard was rebuild after LOST, or after some days of NOQUESTION, it go to CANCELED status
+		CLOSED = 5; // the all work in such contract was done
+	}
+	ContractState state = 2;
+	bytes renter_signature = 3; //renter’s signature for the sub contract
+	bytes host_signature = 4; //host’s signature from the sub contract
+	Timestamp escrow_signed_time = 5; //contract signed time
+	bytes escrow_signature = 6; //escrow's signature
```
Comment (jin-tron) [inline protos/guard/Guard.proto]: This struct is not used in escrow proto. Escrow may not able to sign it for you.

Comment (JustinGaoF) [reply protos/guard/Guard.proto]: It should be the signature which escrow service return at EscrowContractResult, for us know escrow had already work for it.

File: protos/guard/Guard.proto (hunk 44 above)
Comment (taiyangc) [reply protos/guard/Guard.proto]: Still is upper case Guard.proto, maybe needs a git push?

File: protos/core/Btfs.proto (hunk 11 above)
Comment (JustinGaoF) [reply protos/core/Btfs.proto]: removed {}, thanks.

File: protos/core/Btfs.proto (hunk 30 above)
Comment (JustinGaoF) [reply protos/core/Btfs.proto]: done the change.

File: protos/core/Btfs.proto (hunk 32 above)
Comment (JustinGaoF) [reply protos/core/Btfs.proto]: working on it, and gRPC has an embed health endpoint which work with K8s,  working on understanding how to do it. So finally I will remove health related part to another share file.

File: protos/core/Guard.proto (hunk 39 above)
Comment (JustinGaoF) [reply protos/core/Guard.proto]: Yes, will do it

File: protos/guard/guard.proto [hunk 47]
```diff
@@ -0,0 +1,188 @@
...
+    BOOTSTRAP = 2; //reserve
+    PARTIAL_STOP = 3; //reserve field, some function cannot work
+  }
+  HealthStatus status = 3; //current status , running, starting, partial stop
+  google.protobuf.Timestamp curent_time = 4; //current time of guard
+  google.protobuf.Timestamp start_time = 5;
+  bytes git_hash = 6; //the git hash of the build related the git repository's branch
+  bytes version = 7; //version we defined for current node
+  bytes signature = 8; //signature
+}
```
Comment (wxue) [inline protos/guard/guard.proto]: per our sync, it health going to be in a separate PR?  -- I saw its message here, however, rpc service is commented up.

Comment (JustinGaoF) [reply protos/guard/guard.proto]: Yes, it is better to have another PR as they link to different tasks. But currently it is a little complexity for me to separate them as guard.proto should support serverstatus, that's why I combine them now, otherwise we need do a merge.

Comment (JustinGaoF) [reply protos/guard/guard.proto]: exactly

**Comment by wxue** [comment]
haha, can you help add ignore of .DS_Store

**Comment by wxue** [comment]
and delete the previous ones before ignore
